department-of-veterans-affairs/caseflow
# Ryan/appeals 25271 Add filtering to document type

Resolves [Add filtering to document type](https://jira.devops.va.gov/browse/APPEALS-25271)

# Description
This PR includes changes to reader which creates a dropdown for filtering document types. It is identical in functionality to the Issue Tags behavior and also borrows it's data at this point. The Acceptance Criteria seem to be met at this point.

## Acceptance Criteria

## Testing Plan

1. Log into caseflow as BVAAABSHIRE or similar user.
2. Navigate to the Reader portion of caseflow.
3. Click the filter Icon on the documents column.
4. Type in "one", notice that it filters out option "two" but correctly presents "one" and "onee". This indicates that the filter was working as intended.

# Frontend
## User Facing Changes

 BEFORE|AFTER
 
<img width="1325" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/84c9b59c-d5ef-4f54-b9dc-620bdb0c8a80">
|
<img width="952" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110805785/da5c9b39-18e7-45ba-86ef-855b27fdb1e9">
